backend/app/business/open_food_facts/egg_weight_calculator.py

```diff
@@ -55,10 +72,76 @@ def get_total_egg_weight_from_tags(categories_tags: List[str]) -> float:
     """
     num_eggs = get_number_of_eggs(categories_tags)
     weight_per_egg = get_egg_weight_by_tag(categories_tags)
+    if num_eggs > 0 and weight_per_egg == 0:
+        return num_eggs * AVERAGE_EGG_WEIGHT
     return weight_per_egg * num_eggs
```

Comment:
This is mostly a matter of style, but as we're handling more corner cases, it might be clearer to separate them explicitly like this:

```suggestion
    if num_eggs == 0:
        return 0

    if weight_per_egg > 0:
        return num_eggs * weight_per_egg

    # Default to average weight if no specific weight is found
    return num_eggs * AVERAGE_EGG_WEIGHT
```